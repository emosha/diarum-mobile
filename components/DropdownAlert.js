import React from 'react';
import PropTypes from 'prop-types';
import Alert from 'react-native-dropdownalert';

export default class DropdownAlert extends React.Component {
  static propTypes = {
    showAlert: PropTypes.bool.isRequired,
    options: PropTypes.shape({
      type: PropTypes.string,
      title: PropTypes.string,
      message: PropTypes.string,
      interval: PropTypes.number,
    }).isRequired,
  }

  constructor(props) {
    super(props);

    this.dropdown = React.createRef();
  }

  componentDidUpdate() {
    if (this.props.showAlert) {
      this.showAlert(this.props.options);
    }
  }

  /**
   * Triggers allert
   *
   * @param {object} options
   *
   * @returns {void}
   */
  showAlert = ({
    type, title, message, interval = 4000,
  }) => {
    this.dropdown.current.alertWithType(type, title, message, interval);
  }

  render() {
    return (
      <Alert
        ref={this.dropdown}
        testID="dropdown-alert"
      />
    );
  }
}
