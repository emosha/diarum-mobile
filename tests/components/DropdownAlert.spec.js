import React from 'react';
import DropdownAlert from '../../components/DropdownAlert';

describe('DropdownAlert', () => {
  let Component;
  let instance;
  const alertWithType = jest.fn();
  const props = {
    showAlert: false,
    options: {
      type: 'error',
      title: 'Error',
      message: 'an error',
      interval: 2000,
    },
  };

  beforeAll(() => {
    Component = mount(<DropdownAlert {...props} />);
    instance = Component.instance();
    instance.dropdown.current = { alertWithType };
  });

  it('renders correctly', () => {
    expect(Component).toMatchSnapshot();
  });

  it('calls alertWithType when showAlert prop updates to true', () => {
    Component.setProps({
      showAlert: true,
    });

    expect(alertWithType).toHaveBeenCalledTimes(1);
    expect(alertWithType).toHaveBeenCalledWith('error', 'Error', 'an error', 2000);
  });

  it('sets alert interval to 4000 when no interval is provided', () => {
    Component.setProps({
      showAlert: true,
      options: {
        type: 'error',
        title: 'Error',
        message: 'an error',
      },
    });

    expect(alertWithType).toHaveBeenCalledTimes(2);
    expect(alertWithType).toHaveBeenCalledWith('error', 'Error', 'an error', 4000);
  });

  it('does not call alertWithType again when showAlert prop updates to false', () => {
    expect(alertWithType).toHaveBeenCalledTimes(2);

    Component.setProps({
      showAlert: false,
    });

    expect(alertWithType).toHaveBeenCalledTimes(2);
  });
});
