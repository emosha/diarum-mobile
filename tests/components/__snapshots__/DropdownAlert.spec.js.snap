// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownAlert renders correctly 1`] = `
<DropdownAlert
  options={
    Object {
      "interval": 2000,
      "message": "an error",
      "title": "Error",
      "type": "error",
    }
  }
  showAlert={false}
>
  <DropdownAlert
    accessible={false}
    activeStatusBarBackgroundColor="black"
    activeStatusBarStyle="light-content"
    cancelBtnImageSrc={1}
    cancelBtnImageStyle={
      Object {
        "alignSelf": "center",
        "height": 36,
        "padding": 8,
        "width": 36,
      }
    }
    closeInterval={4000}
    containerStyle={
      Object {
        "flexDirection": "row",
        "padding": 16,
      }
    }
    defaultContainer={
      Object {
        "flexDirection": "row",
        "padding": 8,
        "paddingTop": 20,
      }
    }
    defaultTextContainer={
      Object {
        "flex": 1,
        "padding": 8,
      }
    }
    elevation={1}
    endDelta={0}
    errorColor="#cc3232"
    errorImageSrc={1}
    imageSrc={null}
    imageStyle={
      Object {
        "alignSelf": "center",
        "height": 36,
        "padding": 8,
        "width": 36,
      }
    }
    inactiveStatusBarBackgroundColor="black"
    inactiveStatusBarStyle="default"
    infoColor="#2B73B6"
    infoImageSrc={1}
    messageNumOfLines={3}
    messageStyle={
      Object {
        "backgroundColor": "transparent",
        "color": "white",
        "fontSize": 14,
        "fontWeight": "normal",
        "textAlign": "left",
      }
    }
    onCancel={null}
    onClose={null}
    panResponderEnabled={true}
    replaceEnabled={true}
    safeAreaStyle={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
    sensitivity={20}
    showCancel={false}
    startDelta={-100}
    successColor="#32A54A"
    successImageSrc={1}
    tapToCloseEnabled={true}
    testID="dropdown-alert"
    titleNumOfLines={1}
    titleStyle={
      Object {
        "backgroundColor": "transparent",
        "color": "white",
        "fontSize": 16,
        "fontWeight": "bold",
        "textAlign": "left",
      }
    }
    translucent={false}
    updateStatusBar={true}
    useNativeDriver={true}
    warnColor="#cd853f"
    warnImageSrc={1}
    wrapperStyle={null}
    zIndex={null}
  />
</DropdownAlert>
`;
